packages/blocks/src/store/selectors.js

```diff
@@ -243,16 +243,30 @@ export function getActiveBlockVariation( state, blockName, attributes, scope ) {
 			const blockType = getBlockType( state, blockName );
 			const attributeKeys = Object.keys( blockType?.attributes || {} );
 			const definedAttributes = variation.isActive.filter(
-				( attribute ) => attributeKeys.includes( attribute )
+				( attribute ) => {
+					// We support nested attribute paths, e.g. `layout.type`.
+					// In this case, we need to check if the part before the
+					// first dot is a known attribute.
+					const topLevelAttribute = attribute.split( '.' )[ 0 ];
+					return attributeKeys.includes( topLevelAttribute );
+				}
 			);
 			if ( definedAttributes.length === 0 ) {
 				return false;
 			}
-			return definedAttributes.every(
-				( attribute ) =>
-					attributes[ attribute ] ===
-					variation.attributes[ attribute ]
-			);
+			return definedAttributes.every( ( attribute ) => {
+				const attributeValue = getValueFromObjectPath(
+					attributes,
+					attribute
+				);
+				if ( attributeValue === undefined ) {
```

Comment:
Why do we need this check?

Reply:
Yeah, exactly. While we're [verifying that the top-level attribute exists](https://github.com/WordPress/gutenberg/pull/62088/files#diff-7b33304a2d8a4d285862a54312849ae44dd6a8fc179721d444a42c83ec77fa6fR247-R250), we don't do that for any of the nested parts of the path. (I don't think that would be possible, as a `block.json` typically specifies `"type": "object"`, but not necessarily the shape of that object.)

`getValueFromObjectPath` returns `undefined` if an object path isn't found (and if no default value is provided). I couldn't think of a case where a nested attribute should be `undefined`, so I thought it we could assume that it meant that the path wasn't found 🤔